webservice/webservice.py

```diff
@@ -25,9 +27,33 @@
 # how-to-write-docstring-for-url-parameters
 
 
+def authorize_with_dashboard(f):
+    @functools.wraps(f)
+    def check_auth(*args, **kwargs):
+        headers = {}
+        # look at header for what we need
+        if 'Authorization' in request.headers:
+            headers['Authorization'] = request.headers['Authorization']
+        # add cookie to request for auth
+        if 'Cookie' in request.headers:
+            headers['Cookie'] = request.headers['Cookie']
+        # set the X-Forwarded-For and User-agent to prevent session protection on dashboard from causing problems
+        headers['X-Forwarded-For'] = request.headers.get('X-Forwarded-For', request.remote_addr)
+        if 'User-Agent' in request.headers:
+            headers['User-Agent'] = request.headers['User-Agent']
+        # make call to dashboard
+        res = requests.get(request.url_root + 'authorization', headers=headers)
+        if res.status_code == 204:
+            return f(*args, **kwargs)
+        else:
+            return 'Error with authorization: {}'.format(res.text), res.status_code
```

Comment:
I don't think you should provide more details and just return a plain 403 or 401, assuming that's all that ever comes back.

Reply:
Ok will do!